tests/integration_tests.rs

```diff
@@ -1779,7 +1779,7 @@ fn file_with_invalid_utf8_filename() {
         .arg(file_path.as_os_str())
         .assert()
         .success()
-        .stdout("dummy content\n");
+        .stdout("   1 dummy content\n");
```

Comment:
I'm curious, is this really the expected behavior? The test invokes `bat` without any line number style component arguments. I think it should stay as it was before - according to the issue being solved, line numbers should only be displayed when piping when the command line argument was explicitly passed, no?

Reply:
good catch!

I added a check for handle [here](https://github.com/sharkdp/bat/pull/2983/files#diff-64932fd51f70fd457567303b06d7e4e92efb03febb303d5a1e0ba7ac82d4014dR168)

Reply:
this doesn't pass the test now, it seems like there may be an issue with the filename and the is_terminal() function or something like that

> fn is_terminal(&self) -> bool
> 
> ───────────────────────────────────────────
> Returns `true` if the descriptor/handle refers to a terminal/tty.
> 
> On platforms where Rust does not know how to detect a terminal yet, this will return
`false`. ***This will also return `false` if an unexpected error occurred, such as from
passing an invalid file descriptor.***

Reply:
debugging it shows self.config.style_components.numbers() returns true, is there a different way to know if line numbers are requested that i'm missing?

Reply:
My guess would be the following:

The `SimplePrinter` is used whenever stdout is not interactive (so it's paging), `--color` is not set to `always`, `--decorations` is not set to `always` and `--force-colorization` is not set either (see [app.rs](https://github.com/sharkdp/bat/blob/b7e44c76dc7bfe1d4bf6693ac20abdbbc48754b9/src/bin/bat/app.rs#L225-L232), [config.rs](https://github.com/sharkdp/bat/blob/b7e44c76dc7bfe1d4bf6693ac20abdbbc48754b9/src/config.rs#L53-L54) and [controller.rs](https://github.com/sharkdp/bat/blob/b7e44c76dc7bfe1d4bf6693ac20abdbbc48754b9/src/controller.rs#L177-L187)). `--style` controls which style components (such as header or line numbers) are used.  The default for `--style` is `changes,grid,header-filename,numbers,snip`. This default currently stays the same even if `loop_through == true` (which means that `SimplePrinter` will be used). Currently, `SimplePrinter` just does not care about `--style`.

So, when `loop_through == true`, the default for `--style` should be `plain` instead. Then your changes should work since they do when you specify `--style=plain` explicitly.

Reply:
Thank you for the tip @einfachIrgendwer0815, I updated the code now and it's working :)